docs/guides/api-with-nextjs.md

```diff
@@ -0,0 +1,512 @@
+
+    const taskList = await taskListRef.get();
+    const tasks = await query.fetch();
+
+    ctx.res.json({
+      ...taskList,
+      tasks: tasks.documents
+        .map((doc) => ({ id: doc.id, ...doc.content }))
+        .sort(sortByCreatedAt),
+    });
+  } catch (err) {
+    console.log(err);
+    ctx.res.body = 'Failed to retrieve tasks';
+    ctx.res.status = 400;
+  }
+
+  return ctx;
+});
+```
+
+Within our `GET /:listid/:id` endpoint we can start retrieving specific tasks from specific lists.
+
+```typescript
+//apis/tasks-api/functions/tasks.ts
+taskListApi.get('/:listid/:id', async (ctx) => {
+  const { listid, id } = ctx.req.params;
+
+  try {
+    // Get our task list with id [listId]
+    const taskListRef = taskListCol.doc(listid);
+    // Get all tasks from the collection with id [id]
+    const task = await taskListRef.collection<Task>('tasks').doc(id).get();
+
+    ctx.res.json(task);
+  } catch (err) {
+    console.log(err);
+    ctx.res.body = 'Failed to retrieve tasks';
+    ctx.res.status = 400;
+  }
+
+  return ctx;
+});
+```
+
+Within the `PATCH :listid/:id` patch route we can write the logic to update whether a task has been completed.
+
+```typescript
+//apis/tasks-api/functions/tasks.ts
+taskListApi.patch('/:listid/:id', async (ctx) => {
+  const { listid: listId, id } = ctx.req.params;
+  const { completed } = ctx.req.json() as ToggleRequest;
+
+  try {
+    const taskListRef = taskLists.doc(listId);
+    const taskRef = taskListRef.collection<Task>('tasks').doc(id);
+    const originalTask = await taskRef.get();
+
+    await taskListRef
+      .collection<Task>('tasks')
+      .doc(id)
+      .set({
+        ...originalTask,
+        complete: completed,
+      });
+
+    ctx.res.body = 'Successfully updated task';
+  } catch (err) {
+    console.log(err);
+    ctx.res.body = 'Failed to retrieve tasks';
+    ctx.res.status = 400;
+  }
+
+  return ctx;
+});
+```
+
+The delete routes `DELETE /:listid/:id` and `DELETE /:id` get the relevent document and delete them from the collection.
+
+```typescript
+//apis/tasks-api/functions/tasks.ts
+taskListApi.delete('/:listid/:id', async (ctx) => {
+  const { listid: listId, id } = ctx.req.params;
+
+  try {
+    const taskListRef = taskLists.doc(listId);
+    await taskListRef.collection('tasks').doc(id).delete();
+    ctx.res.body = 'Successfully deleted task';
+  } catch (err) {
+    console.log(err);
+    ctx.res.body = 'Failed to delete task';
+    ctx.res.status = 400;
+  }
+
+  return ctx;
+});
+
+taskListApi.delete('/:id', async (ctx) => {
+  const { id } = ctx.req.params;
+
+  try {
+    await taskLists.doc(id).delete();
+    ctx.res.body = 'Successfully deleted task list';
+  } catch (err) {
+    console.log(err);
+    ctx.res.body = 'Failed to delete task list';
+    ctx.res.status = 400;
+  }
+
+  return ctx;
+});
+```
+
+### Set up API proxy
+
+Start by create your `.env` file by renaming the `.env.example` file:
+
+```
+mv web/apps/.env.example web/apps/.env
```

Comment:
The .env file is checked in to the repo instead of the .env.example

Reply:
I will take over this one and fix it up